### PR TITLE
Update the golang and alpine base images

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The golang base image is now updated to 1.16.15. The alpine base image is updated to 3.16.2.
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ############# golang-base
-FROM golang:1.16.3 AS golang-base
+FROM golang:1.16.15 AS golang-base
```
Comment (ialidzhikov) [inline Dockerfile:6]: Can you also update the `verify` image in the pipeline definition to 1.16.15?

Comment (ialidzhikov) [inline Dockerfile:6]: Nit: you could also keep in sync the go version in https://github.com/gardener/terraformer/blob/72966630f66e43f8d8fd4c4ac16d8c12685a007d/.test-defs/pod-e2e.yaml#L18

Comment (kon-angelo) [reply Dockerfile:6]: Thanks I missed that.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**